packages/hawtio/src/plugins/shared/workspace.ts

```diff
@@ -46,6 +54,68 @@ class Workspace {
     })
   }
 
+  private maybeUpdatePlugins(response: IResponse): void {
+    if (this.pluginUpdateCounter === null) {
+      this.pluginUpdateCounter = response.value as number | null
+      return
+    }
+    if (this.pluginUpdateCounter !== response.value) {
+      if (parseBoolean(localStorage['autoRefresh'])) {
+        window.location.reload()
+      }
+    }
+  }
+
+  private maybeReloadTree(response: IResponse): void {
+    const counter = response.value
+    if (this.treeWatcherCounter === null) {
+      this.treeWatcherCounter = counter as number | null
+      return
+    }
+    if (this.treeWatcherCounter !== counter) {
+      this.treeWatcherCounter = counter as number | null
+      this.refreshTree()
+    }
+  }
+
+  async maybeMonitorPlugins() {
+    const hasRegistry = await this.treeContainsDomainAndProperties('hawtio', { type: 'Registry' })
+    const hasTreeWatcher = await this.treeContainsDomainAndProperties('hawtio', { type: 'TreeWatcher' })
+
+    if (hasRegistry) {
+      if (this.pluginRegisterHandle === null) {
+        this.pluginRegisterHandle = await jolokiaService.register(
+          {
+            type: 'read',
+            mbean: HAWTIO_REGISTRY_MBEAN,
+            attribute: 'UpdateCounter',
+          },
+          (response: IResponse) => this.maybeUpdatePlugins(response),
+        )
+      }
+    } else {
+      if (this.pluginRegisterHandle !== null) {
+        jolokiaService.unregister(this.pluginRegisterHandle)
+        this.pluginRegisterHandle = null
+        this.pluginUpdateCounter = null
+      }
+    }
+
+    // lets also listen to see if we have a JMX tree watcher
+    if (hasTreeWatcher) {
+      if (this.treeWatchRegisterHandle === null) {
+        this.pluginRegisterHandle = await jolokiaService.register(
+          {
+            type: 'read',
+            mbean: HAWTIO_TREE_WATCHER_MBEAN,
+            attribute: 'Counter',
+          },
+          (response: IResponse) => this.maybeReloadTree(response),
+        )
+      }
+    }
```

Comment:
Why doesn't the tree watcher have the same unregistering branch with the plugin registry?  I see the old impl doesn't as well, but is there a good reason not to do so?